src/core/tsi/ssl_transport_security.c

```diff
@@ -846,6 +858,51 @@ static const tsi_frame_protector_vtable frame_protector_vtable = {
     ssl_protector_destroy,
 };
 
+/* --- tsi_server_handshaker_factory methods implementation. --- */
+
+static void tsi_ssl_handshaker_factory_destroy(
+    tsi_ssl_handshaker_factory *self) {
+  GPR_ASSERT(self != NULL);
+  GPR_ASSERT(self->vtable != NULL);
+  if (self->vtable->destroy != NULL) {
+    self->vtable->destroy(self);
+  }
+  /* Note, we don't free(self) here because this object is always directly
+   * embedded in another object. If tsi_ssl_handshaker_factory_init allocates
+   * any memory, it should be free'd here. */
+}
+
+static void tsi_ssl_handshaker_factory_addref(
```

Comment:
The signature for a ref method should be:
static tsi_ssl_handshaker_factory *tsi_ssl_handshaker_factory_ref(tsi_ssl_handshaker_factory *self);